lib/service/supervisor.go

```diff
@@ -75,9 +75,13 @@ type Supervisor interface {
 	RegisterEventMapping(EventMapping)
 
 	// ExitContext returns context that will be closed when
-	// TeleportExitEvent is broadcasted.
+	// a hard TeleportExitEvent is broadcasted.
 	ExitContext() context.Context
```

Comment:
Suggestion:

```suggestion
	HardExitContext() context.Context
```

Reply:
In order to keep consistent with the established scheme of having `Shutdown()` and `Close()` methods for graceful and immediate exits respectively, I've renamed `ExitContext` to `CloseContext` instead and left `ShutdownContext` as-is.